src/main/java/org/terasology/segmentedpaths/controllers/SegmentCacheSystem.java

```diff
@@ -29,13 +29,23 @@
 import java.util.HashMap;
 
 /**
- * Created by michaelpollind on 4/3/17.
+ * Cache system for segments. Constructs, stores and supplies segments based off their prefabs.
  */
 @RegisterSystem(RegisterMode.AUTHORITY)
 @Share(value = SegmentCacheSystem.class)
 public class SegmentCacheSystem extends BaseComponentSystem {
+
+    /**
+    * The map, which stores already constructed segments with their prefab's name as a key.
+    */
     private HashMap<String, Segment> segments = new HashMap<>();
 
+    /**
+     * Gets a segment based off its prefab.
+     * If a segment was already constructed, returns a segment from the cache, otherwise constructs it from a prefab.
```

Comment:
`If a segment was already, returns a segment from the cache`
a segment - any segment? If I request segment `x`, but segment `y` and segment `z` have already been constructed, your wording allows the system to return A segment (either `y` or `z`) since A segment (`y` or `z`) has already been constructed.
`If the specified segment has alredy been constructed..` or `If a segment based on the specified prefab has already been constructed..` or something similar.